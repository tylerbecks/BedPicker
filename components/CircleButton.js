import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";

export const BUTTON_SIZE = 90;

const styles = StyleSheet.create({
  circle: {
    alignItems: "center",
    backgroundColor: "#CFD8DC",
    borderRadius: BUTTON_SIZE,
    elevation: 2, // Android
    height: BUTTON_SIZE,
    justifyContent: "center",
    margin: 10,
    shadowColor: "rgba(0,0,0, .4)", // iOS
    shadowOffset: { height: 1, width: 1 }, // iOS
    shadowOpacity: 1, // iOS
    shadowRadius: 1, // iOS
    width: BUTTON_SIZE
  }
});

const CircleButton = ({ onPress, children, style }) => (
  <TouchableOpacity
    activeOpacity={0.4}
    style={{ ...styles.circle, ...style }}
    onPress={onPress}
  >
    {children}
  </TouchableOpacity>
);

export default CircleButton;
