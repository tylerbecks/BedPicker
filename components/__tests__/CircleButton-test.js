import 'react-native';
import React from 'react';
import CircleButton from '../CircleButton';
import renderer from 'react-test-renderer';

it('renders', () => {
  expect(
    renderer.create(<CircleButton onPress={jest.fn()} />).toJSON()
  ).toMatchSnapshot();
});
