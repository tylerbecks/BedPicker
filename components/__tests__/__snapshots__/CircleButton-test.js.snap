// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#CFD8DC",
      "borderRadius": 90,
      "elevation": 2,
      "height": 90,
      "justifyContent": "center",
      "margin": 10,
      "opacity": 1,
      "shadowColor": "rgba(0,0,0, .4)",
      "shadowOffset": Object {
        "height": 1,
        "width": 1,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
      "width": 90,
    }
  }
/>
`;
